### ISSUE TITLE
SA1515 should not trigger for C# 12 collection expressions

### ISSUE THREAD

**Issue by 50Wliu** [DotNetAnalyzers/StyleCopAnalyzers#3766]
```cs
string[] elements =
[
  // Hydrogen <-- SA1515 Single-line comment should be preceded by blank line
  "H",

  // Helium
  "He",
]
```

This doesn't seem right to me, as the same code using the curly bracket syntax doesn't trigger the warning.

https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions

1.2.0-beta.556
